Download reports only if they're necessary

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119027
* #118931
* #118882
* #118874

Previously we were downloading all of (eager311, dynamo38, dynamo311).
Now we just download what's necessary. This is useful for
update_failures.py because the dynamo tests finish much faster than the
eager tests and it only needs the result from the dynamo tests.

zou3519 committed:
Download reports only if they're necessary

Previously we were downloading all of (eager311, dynamo38, dynamo311).
Now we just download what's necessary. This is useful for
update_failures.py because the dynamo tests finish much faster than the
eager tests and it only needs the result from the dynamo tests.

[ghstack-poisoned]

zou3519 committed:
Update on "Download reports only if they're necessary"


Previously we were downloading all of (eager311, dynamo38, dynamo311).
Now we just download what's necessary. This is useful for
update_failures.py because the dynamo tests finish much faster than the
eager tests and it only needs the result from the dynamo tests.

[ghstack-poisoned]

jamesjwu commented on scripts/compile_tests/download_reports.py:
```diff
@@ -1,57 +1,78 @@
+        missing_configs.append(config)
+    if len(missing_configs) == 0:
+        print(
+            "All required logs appear to exist, not downloading again. rm the log dir if this is not the case"
```
Nit: suggest `rm -r {log_dir}` for a copy pasteable command

```
f"All required logs appear to exist, not downloading again. Run `rm -r {log_dir}` if this is not the case"
```

zou3519 committed:
Update on "Download reports only if they're necessary"


Previously we were downloading all of (eager311, dynamo38, dynamo311).
Now we just download what's necessary. This is useful for
update_failures.py because the dynamo tests finish much faster than the
eager tests and it only needs the result from the dynamo tests.

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge -f "unrelated failure"
